paddle/fluid/framework/ir/seqpool_concat_fuse_pass.cc

```diff
@@ -39,21 +39,25 @@ PDNode* BuildSeqPoolConcatPattern(PDPattern* pattern,
 
   auto is_seqpool_op_with_pootype_of_nth_input_of_concat = [=](
       Node* x, const std::string& type, int idx) -> bool {
-    bool ok = x && x->IsOp() && x->Op()->Type() == "sequence_pool" &&
-              x->Op()->HasAttr("pooltype") &&
-              boost::get<std::string>(x->Op()->GetAttr("pooltype")) == type &&
-              x->outputs.size() == 2;  // seqpool should only have 2 outputs
-    if (ok) {
-      // only one output of seqpool_op is nth_input_var of concat
-      // the other one should be unused empty var
+    bool this_is_seqpool_op =
+        x && x->IsOp() && x->Op()->Type() == "sequence_pool" &&
```

Comment:
`x && x->IsOp() && x->Op()` ...

Reply:
Thanks, I know your concern, but I thought this is fine since : 
 https://github.com/PaddlePaddle/Paddle/blob/8f17c714de6e2a99c6f7ebf1dc895d65047b372a/paddle/fluid/framework/ir/graph_pattern_detector.cc#L594-L597

https://github.com/PaddlePaddle/Paddle/blob/8f17c714de6e2a99c6f7ebf1dc895d65047b372a/paddle/fluid/framework/ir/graph_pattern_detector.cc#L371-L375